day2/day2.erl

```diff
@@ -0,0 +1,59 @@
+-module(day2).
+-export([test_new_computer/0, find_input_pair_checksum/0]).
+
+read_input(FileName) ->
+    {ok, Data} = file:read_file(FileName),
+    binary:split(Data, [<<",">>], [global]).
+
+process_input(L) ->
+    [ list_to_integer(binary_to_list(X)) || X <- lists:append(lists:sublist(L, length(L) - 1), [binary:replace(lists:nth(length(L), L), <<"\n">>, <<"">>)]) ].
```

Comment:
An alternative would be to store a list of `[{Position, Data}]` so `1,3,4,5,99` would become `[{0,1}, {1,3}, {2,4}, {3,5}, {4,99}]`. Then you can look at using the `lists:key*` functions like keyreplace, keyfind, etc. Then you can stay true to the 0-indexed nature of the Intcode processor :)

Reply:
I spent a lot of time working on this problem today. I solved it using a list of tuples and maps. When using maps it felt strange having to heavily relay on BIF rather than pattern matching. My current solution (which was heavily influenced by your intcode computer) uses a combination of both methods but I'm assuming it is best practice to use `maps:get` to extract values from maps in all cases instead of mixing the two.